protocol/x/vault/app_test.go

```diff
@@ -0,0 +1,326 @@
+package vault_test
+
+import (
+	"bytes"
+	"math/big"
+	"testing"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/dtypes"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+	vaulttypes "github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+	"github.com/stretchr/testify/require"
+)
+
+// DepositInstance represents an instance of a deposit to test.
+type DepositInstance struct {
+	// Depositor.
+	depositor satypes.SubaccountId
+	// Amount to deposit (in quote quantums).
+	depositAmount *big.Int
+	// Signer of the message.
+	msgSigner string
+
+	// A string that CheckTx response should contain, if any.
+	checkTxResponseContains string
+	// Whether CheckTx fails.
+	checkTxFails bool
+	// Whether DeliverTx fails.
+	deliverTxFails bool
+}
+
+// DepositorSetup represents the setup of a depositor.
+type DepositorSetup struct {
+	// Depositor.
+	depositor satypes.SubaccountId
+	// Initial balance of the depositor (in quote quantums).
+	depositorBalance *big.Int
+}
+
+func TestMsgDepositToVault(t *testing.T) {
```

Comment:
The usual convention in this code base is that `file_name_test.go` is used to test `file_name.go`. Is this file supposed to test app.go code in vault?

Reply:
[see answer above](https://github.com/dydxprotocol/v4-chain/pull/1220#discussion_r1536562209)